Sources/Controllers/BasicAudioController.swift

```diff
@@ -0,0 +1,223 @@
+
+    /// The `Timer` that update playing progress
+    internal var progressTimer: Timer?
+
+    // MARK: - Init Methods
+
+    public init(messageCollectionView: MessagesCollectionView) {
+        self.messageCollectionView = messageCollectionView
+        super.init()
+    }
+
+    // MARK: - Methods
+
+    /// Used to configure the audio cell UI:
+    ///     1. play button selected state;
+    ///     2. progresssView progress;
+    ///     3. durationLabel text;
+    ///
+    /// - Parameters:
+    ///   - cell: The `AudioMessageCell` that needs to be configure.
+    ///   - message: The `MessageType` that configures the cell.
+    ///
+    /// - Note:
+    ///   This protocol method is called by MessageKit every time an audio cell needs to be configure
+    open func configureCell(_ cell: AudioMessageCell, message: MessageType) {
+        if playingMessage?.messageId == message.messageId, let collectionView = messageCollectionView, let player = audioPlayer {
+            // cell needs to be configure since the current audio player is playing sound for receive message
+            // This situation can happen when playing cell is reconfigured
+            playingCell = cell
+            cell.progressView.progress = (player.duration == 0) ? 0 : Float(player.currentTime/player.duration)
+            cell.playButton.isSelected = (player.isPlaying == true) ? true : false
+            guard let displayDelegate = collectionView.messagesDisplayDelegate else {
+                fatalError(MessageKitError.nilMessagesDisplayDelegate)
+            }
+            cell.durationLabel.text = displayDelegate.audioProgressTextFormat(Float(player.duration), for: cell, in: collectionView)
+        }
+    }
+
+    /// Used to start play audio sound
+    ///
+    /// - Parameters:
+    ///   - message: The `MessageType` that contain the audio item to be played.
+    ///   - audioCell: The `AudioMessageCell` that needs to be updated while audio is playing.
+    open func playSound(for message: MessageType, in audioCell: AudioMessageCell) {
+        switch message.kind {
+        case .audio(let item):
+            playingCell = audioCell
+            playingMessage = message
+            guard let player = try? AVAudioPlayer(contentsOf: item.url) else {
+                print("Failed to create audio player for URL: \(item.url)")
+                return
+            }
+            audioPlayer = player
+            audioPlayer?.prepareToPlay()
+            audioPlayer?.delegate = self
+            audioPlayer?.play()
+            state = .playing
+            audioCell.playButton.isSelected = true  // show pause button on audio cell since the sound is currently playing
+            startProgressTimer()
+            // call start audio delegate
+            audioCell.delegate?.didStartAudio(in: audioCell)
+        default:
+            print("Error - can not play sound for a message that is not of kind .audio")
```

Comment:
I believe we have a file for MessageKit errors, can you please move this there and follow the convention we use for printing errors?